File: packages/jsii-pacmak/test/expected.dotnet/AWS.Jsii.Tests.Calculator/JSObjectLiteralForInterface.cs
using AWS.Jsii.Runtime.Deputy;
using AWS.Jsii.Tests.Calculator.Lib;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass("jsii-calc", "jsii$jsii_calc$.JSObjectLiteralForInterface", "[]")]
    public class JSObjectLiteralForInterface : DeputyBase
    {
        public JSObjectLiteralForInterface(): base(new DeputyProps(new object[]{}))
        {
        }

        protected JSObjectLiteralForInterface(ByRefValue reference): base(reference)
        {
        }

        protected JSObjectLiteralForInterface(DeputyProps props): base(props)
        {
        }

        [JsiiMethod("giveMeFriendly", "{\"fqn\":\"jsii$jsii_calc_lib$.IFriendly\"}", "[]")]
        public virtual IIFriendly GiveMeFriendly()
        {
            return InvokeMethod<IIFriendly>(new object[]{});
        }

        [JsiiMethod("giveMeFriendlyGenerator", "{\"fqn\":\"jsii$jsii_calc$.IFriendlyRandomGenerator\"}", "[]")]
        public virtual IIFriendlyRandomGenerator GiveMeFriendlyGenerator()
        {
            return InvokeMethod<IIFriendlyRandomGenerator>(new object[]{});
        }
    }
}

File: packages/jsii-pacmak/test/expected.java.jsii-calc/java/org/jsii/tests/calculator/JSObjectLiteralForInterface.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.JSObjectLiteralForInterface")
public class JSObjectLiteralForInterface extends org.jsii.JsiiObject {
    protected JSObjectLiteralForInterface(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public JSObjectLiteralForInterface() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public org.jsii.tests.calculator.lib.IFriendly giveMeFriendly() {
        return this.jsiiCall("giveMeFriendly", org.jsii.tests.calculator.lib.IFriendly.class);
    }
    public org.jsii.tests.calculator.IFriendlyRandomGenerator giveMeFriendlyGenerator() {
        return this.jsiiCall("giveMeFriendlyGenerator", org.jsii.tests.calculator.IFriendlyRandomGenerator.class);
    }
}
